### PR TITLE
Add external service settings

### PR DESCRIPTION
**Pull request by lucasxciv**
## Summary

Add routes to handle external service settings. 

This implementation will include only the `quickBooksOnline` service.

Screenshot:

Screenshot 2023-05-16 183927

## Links

Depends on https://github.com/Rebilly/api-definitions/pull/1379

## Checklist

- [X] Writing style
- [X] API design standards

### CODE REVIEW

File: openapi/components/schemas/ExternalServiceSettings.yaml [hunk 1]
```diff
@@ -0,0 +1,10 @@
+type: object
+description: Settings for external service integration.
+properties:
+  quickBooksOnline:
+    type: object
+    description: QuickBooks Online settings.
+    properties:
+      syncJournalEntries:
+        type: boolean
+        description: Enable or disable journal entries synchronization.
```
Comment (ninjasimon) [inline openapi/components/schemas/ExternalServiceSettings.yaml:10]:
```suggestion
        description: Specifies if journal entries should be synchronized.
```
https://github.com/Rebilly/api-definitions/blob/7b8746ac0d49f1c59946a9abd695c5ad22ebc513/WRITING-STYLE.md?plain=1#L255

File: openapi/paths/external-services-settings.yaml [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+      content:
+        application/json:
+          schema:
+            $ref: ../components/schemas/ExternalServiceSettings.yaml
+    '401':
+      $ref: ../components/responses/Unauthorized.yaml
+    '403':
+      $ref: ../components/responses/Forbidden.yaml
+    '404':
+      $ref: ../components/responses/NotFound.yaml
```
Comment (justnero) [inline openapi/paths/external-services-settings.yaml:22]: I don't think there could be a 404
```suggestion
```

File: openapi/paths/external-services-settings.yaml [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
```
Comment (justnero) [inline openapi/paths/external-services-settings.yaml:50]: Same as above
```suggestion
```

File: openapi/paths/external-services-settings.yaml (hunk 2 above)
Comment (lucasxciv) [reply openapi/paths/external-services-settings.yaml]: That's correct. Fixed. The default will be `false`.

File: openapi/components/schemas/ExternalServiceSettings.yaml [hunk 4]
```diff
@@ -0,0 +1,10 @@
+type: object
+description: Settings for external service integration.
+properties:
+  quickBooksOnline:
+    type: object
+    description: QuickBooks Online settings.
+    properties:
+      syncJournalEntries:
+        type: boolean
+        description: Specifies if journal entries should be synchronized.
```
Comment (justnero) [inline openapi/components/schemas/ExternalServiceSettings.yaml:10]:
```suggestion
        description: Specifies if journal entries should be synchronized.
        default: false
```

**Review by justnero — DISMISSED** [review]

File: openapi/components/schemas/ExternalServiceSettings.yaml [hunk 5]
```diff
@@ -0,0 +1,11 @@
...
```
Comment (noon-dawg) [inline openapi/components/schemas/ExternalServiceSettings.yaml:10]: nitpick: avoid using [should](https://github.com/Rebilly/api-definitions/blob/main/WRITING-STYLE.md#general-guidance)
```suggestion
        description: Specifies if journal entries must be synchronized.
```

**Review by noon-dawg — APPROVED** [review]
Looks good on the docs side!

**Review by justnero — APPROVED** [review]

**Review by ninjasimon — APPROVED** [review]

**Review by cesarlevel — APPROVED** [review]

**Pull request merged**